internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -903,31 +969,177 @@ func (r *AgentServiceConfigReconciler) newImageServiceDeployment(ctx context.Con
 	}
 
 	mutateFn := func() error {
-		if err := controllerutil.SetControllerReference(instance, deployment, r.Scheme); err != nil {
+		if err := controllerutil.SetControllerReference(instance, statefulSet, r.Scheme); err != nil {
 			return err
 		}
 		var replicas int32 = 1
-		deployment.Spec.Replicas = &replicas
+		statefulSet.Spec.Replicas = &replicas
 
-		deployment.Spec.Template.Spec.Containers = []corev1.Container{container}
-		deployment.Spec.Template.Spec.Volumes = volumes
-		deployment.Spec.Template.Spec.ServiceAccountName = imageServiceName
+		statefulSet.Spec.Template.Spec.Containers = []corev1.Container{container}
+		statefulSet.Spec.Template.Spec.ServiceAccountName = imageServiceName
+
+		volumes := ensureVolume(statefulSet.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: "tls-certs",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: imageServiceName,
+				},
+			},
+		})
+		volumes = ensureVolume(volumes, corev1.Volume{
+			Name: "service-cabundle",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: imageServiceName,
+					},
+				},
+			},
+		})
+
+		if instance.Spec.ImageStorage != nil {
+			var found bool
+			for i, claim := range statefulSet.Spec.VolumeClaimTemplates {
+				if claim.ObjectMeta.Name == "image-service-data" {
+					found = true
+					statefulSet.Spec.VolumeClaimTemplates[i].Spec.Resources.Requests = getStorageRequests(instance.Spec.ImageStorage)
+				}
+			}
+			if !found {
+				statefulSet.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
```

Comment:
Could we add a label here to make it easier to identify these PVCs as belonging to this particular statefulset?

Reply:
It actually looks like the pvcs are inheriting the labels from the statefulset so we can probably just use `app: assisted-image-service`

Probably better than what I have here https://github.com/openshift/assisted-service/pull/3067/files/b206f2e1b8b8e4d1a5ab1c3605e324651a665193#diff-c619bb36181b5e1dd76fb1fcc1f54068b16a8761f77106052fcbf5e2f5f9d356R860 :sweat_smile:

Reply:
Yeah, and will enable cleanup if/when the pods are gone.